framework/Models/PostProcessors/DataClassifier.py

```diff
@@ -93,133 +93,93 @@ def _handleInput(self, paramInput):
-          else:
-            self.raiseAnError(IOError, "None of the input DataObjects can be used as the reference classifier! Either the label", \
-                    self.label, "is not exist in the output of the DataObjects or the inputs of the DataObjects are not the same as", \
-                    ','.join(self.mapping.keys()))
-        newInput[dataType]['input'] = dict.fromkeys(inputParams)
-        newInput[dataType]['output'] = dict.fromkeys(outputParams)
-        if inputObject.type == 'PointSet':
-          for elem in inputParams:
-            newInput[dataType]['input'][elem] = copy.deepcopy(inputDataset[elem].values)
-          for elem in outputParams:
-            newInput[dataType]['output'][elem] = copy.deepcopy(inputDataset[elem].values)
-          newInput[dataType]['type'] = inputObject.type
-          newInput[dataType]['name'] = inputObject.name
+          self.raiseAnError(IOError, "Both input data objects have been already processed! No need to execute this postprocessor", self.name)
+        if inputDict['type'] != 'PointSet':
+          self.raiseAnError(IOError, "Only PointSet is allowed as classifier, but got", inputDict['type'])
+      else:
+        dataType = 'target'
+        if not haveTarget:
+          haveTarget = True
         else:
-          # only extract the last element in each realization for the HistorySet
-          newInput[dataType]['type'] = inputObject.type
-          newInput[dataType]['name'] = inputObject.name
-          numRlzs = len(inputObject)
-          newInput[dataType]['historySizes'] = dict.fromkeys(range(numRlzs))
-          for i in range(numRlzs):
-            rlz = inputObject.realization(index=i)
-            for elem in inputParams:
-              if newInput[dataType]['input'][elem] is None:
-                newInput[dataType]['input'][elem] = np.empty(0)
-              newInput[dataType]['input'][elem] = np.append(newInput[dataType]['input'][elem], rlz[elem])
-            for elem in outputParams:
-              if newInput[dataType]['output'][elem] is None:
-                newInput[dataType]['output'][elem] = np.empty(0)
-              newInput[dataType]['output'][elem] = np.append(newInput[dataType]['output'][elem], rlz[elem].values[-1])
-              if newInput[dataType]['historySizes'][i] is None:
-                newInput[dataType]['historySizes'][i] = len(rlz[elem].values)
-
+          self.raiseAnError(IOError, "None of the input DataObjects can be used as the reference classifier! Either the label", \
+                  self.label, "is not exist in the output of the DataObjects or the inputs of the DataObjects are not the same as", \
+                  ','.join(self.mapping.keys()))
+      newInput[dataType] = inputDict
     return newInput
 
   def run(self, inputIn):
     """
       This method executes the postprocessor action.
-      @ In,  inputIn, list, list of DataObjects
+      @ In,  inputIn, list, list of input dictionaries
```

Comment:
please define what those dictionaries look like (what do we have inside?)

Reply:
e.g. ```dims```, ```type```

Reply:
updated